strawberry/cli/commands/codegen.py

```diff
@@ -143,12 +178,50 @@ def codegen(
     ),
     cli_plugin: Optional[str] = None,
 ) -> None:
+    if not query:
+        return
+
     schema_symbol = load_schema(schema, app_dir)
 
-    console_plugin = _load_plugin(cli_plugin) if cli_plugin else ConsolePlugin
+    console_plugin_type = _load_plugin(cli_plugin) if cli_plugin else ConsolePlugin
 
-    plugins = _load_plugins(selected_plugins)
-    plugins.append(console_plugin(query, output_dir, plugins))
+    plugins = _load_plugins(selected_plugins, query[0])
+    console_plugin = console_plugin_type(query[0], output_dir, plugins)
```

Comment:
I'm not sure I like having to pass the first query here 🤔

Reply:
Yeah, I dislike this as well.  I'd rather have the `console_plugin_type` accept a collection of queries, but anyone who has made a custom console plugin type would potentially be broken in that case :(.